src/storage/exec/IndexOutputNode.h

```diff
@@ -219,27 +218,26 @@ class IndexOutputNode final : public RelNode<T> {
         auto returnCols = result_->colNames;
         for (const auto& val : data) {
             Row row;
-            auto src = NebulaKeyUtils::getSrcId(planContext_->vIdLen_, val.first);
-            auto rank = NebulaKeyUtils::getRank(planContext_->vIdLen_, val.first);
-            auto dst = NebulaKeyUtils::getDstId(planContext_->vIdLen_, val.first);
-            if (planContext_->isIntId_) {
-                row.emplace_back(src.toString());
-                row.emplace_back(rank);
-                row.emplace_back(dst.toString());
-            } else {
-                row.emplace_back(src.subpiece(0, src.find_first_of('\0')).toString());
-                row.emplace_back(rank);
-                row.emplace_back(dst.subpiece(0, dst.find_first_of('\0')).toString());
-            }
             auto reader = RowReaderWrapper::getRowReader(schema, val.second);
             if (!reader) {
                 VLOG(1) << "Can't get tag reader";
                 return kvstore::ResultCode::ERR_EDGE_NOT_FOUND;
             }
-            // skip column src_, ranking, dst_
-            for (size_t i = 3; i < returnCols.size(); i++) {
-                auto v = reader->getValueByName(returnCols[i]);
-                row.emplace_back(std::move(v));
+            for (const auto& colName : returnCols) {
+                if (colName == kSrc) {
+                    auto src = NebulaKeyUtils::getSrcId(planContext_->vIdLen_, val.first);
+                    addVidToRow(src, row);
+                } else if (colName == kType) {
+                    row.emplace_back(planContext_->edgeType_);
+                } else if (colName == kRank) {
+                    row.emplace_back(NebulaKeyUtils::getRank(planContext_->vIdLen_, val.first));
+                } else if (colName == kDst) {
+                    auto dst = NebulaKeyUtils::getDstId(planContext_->vIdLen_, val.first);
+                    addVidToRow(dst, row);
+                } else {
+                    auto v = reader->getValueByName(colName);
+                    row.emplace_back(std::move(v));
+                }
```

Comment:
Why need to collect TagId and EdgeType in index result set?  I guess the tagId and EdgeType are used for pipe . right?

Reply:
Yeah, usually we don't need it, if query engine specify it, need to return it as well..